### PR TITLE
Sync connection parameter bundleId

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

File: packages/realm/bindgen/vendor/realm-core
[the remark this answers is not in the document]
Comment (kneth) [reply packages/realm/bindgen/vendor/realm-core:1]: Yes, I needed to expose `bundle_id`

File: packages/realm/src/platform/node/device-info.ts [hunk 1]
```diff
@@ -17,8 +17,12 @@
 ////////////////////////////////////////////////////////////////////////////
 
 import os from "node:os";
+import process from "node:process";
 
 import { version } from "realm/package.json";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (elle-j) [inline packages/realm/src/platform/node/device-info.ts:24]: What error are you getting without these? Looks like TS doesn't yell for the line right above it.

File: packages/realm/src/platform/react-native/device-info.ts [hunk 2]
```diff
@@ -19,6 +19,9 @@
 import { Platform } from "react-native";
 
 import { version } from "realm/package.json";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (elle-j) [inline packages/realm/src/platform/react-native/device-info.ts:23]: I've got the same question as the previous one 🙂

File: packages/realm/src/platform/node/device-info.ts (hunk 1 above)
Comment (kneth) [reply packages/realm/src/platform/node/device-info.ts:24]: Yeah, you might be right. It was yelling when I used a JavaScript file but switching to JSON might have helped.

File: packages/realm/src/platform/react-native/device-info.ts (hunk 2 above)
Comment (kneth) [reply packages/realm/src/platform/react-native/device-info.ts]: Same answer

**Comment by kraenhansen** [comment]
I'd suggest writing to the `realm/package.json` instead - that's guaranteed to always be present and therefore we can rely on it being there and not have to handle the special case of the postinstall script failing to produce the file (if #6038 gets merged and analytics is disabled when installing).

**Pull request merged**